### DISCUSSION TITLE
Add git remote-url property

### DISCUSSION THREAD

**Discussion by cadayton** [jdhitsolutions/PSProjectStatus#4] [Ideas]
Many of my projects can live in different remote repositories.  For instance, GitHub, GitLab, KeyBase, and maybe IPFS.

It would be desirable to track the remoteURL of each project along with it's status.

ex.
PS>git remote -v
origin  keybase://team/psmodules/PSKeyBase (fetch)
origin  keybase://team/psmodules/PSKeyBase (push)

Thanks.  Liking the PSProjectStatus module.

**Comment by jdhitsolutions** [comment]
So all those repositories are still using git under the hood? I'm assuming getting the remotes with push is of more importance than fetch.

**Comment by cadayton** [comment]
Yes. All repositories are using git. It would just be nice to filter through projects based where the remote repository is located.

Craig Dayton
***@***.******@***.***)m.me
360-466-8778
[https://cadayton.keybase.pub](https://cadayton.keybase.pub//)

Sent with [ProtonMail](https://protonmail.com/) secure email.

------- Original Message -------

**Comment by jdhitsolutions** [comment]
Saving this code snippet for future use.

```powershell
git remote show | foreach { git remote get-url --push $_}
```

**Reply by cadayton** [reply]
![image](https://user-images.githubusercontent.com/5587738/159188429-b954d2d9-68b4-400c-95ed-df908e1305ee.png)

I tried creating a branch to the PSProjectStatus but permissions wouldn't let me so I [fork it](https://github.com/cadayton/PSProjectStatus).

Add 'Version' and 'SVC' property plus add logical for SVN.  I didn't change the format typing to handle the new properties yet.

In the sub directory, 'podeweb' is the code necessary to implement the Pode.Web server and the page being displayed here.  The projects folders are hard coded into the page logic so this needs to be updated to suit each implementation.

**Comment by jdhitsolutions** [comment]
I'm adding a project version value pulled from a module manifest if found as well as the remote repo information.

![image](https://user-images.githubusercontent.com/12089920/159744716-9024bc6a-5cc1-4e27-b842-e3d7de3f63f0.png)

![image](https://user-images.githubusercontent.com/12089920/159746255-dee06a4d-8360-4253-a589-7ad74f398519.png)

**Comment by jdhitsolutions** [comment]
I've published an update that adds new properties. Update to v0.5.0 and look at the `README` file.

**Comment by cadayton** [comment]
Thanks, Jeff! I'll check it out.

Craig Dayton
***@***.******@***.***)m.me
360-466-8778
[https://cadayton.keybase.pub](https://cadayton.keybase.pub//)

Sent with [ProtonMail](https://protonmail.com/) secure email.

------- Original Message -------

**Comment by cadayton** [comment]
Hi Jeff,

If your are interesting in putting a web interface on PSProjectStatus using Pode.Web, here is an example of how I did something similar.

https://cadayton.keybase.pub/PSGallery/Pode/PW-KVExplorer.html

Go here to see a history of shared stuff I've done in PowerShell.

https://cadayton.keybase.pub/PowerShell

------- Original Message -------
